mobile/src/main/java/ch/epfl/sdp/mobile/ui/game/ar/ChessScene.kt

```diff
@@ -72,90 +90,47 @@ class ChessScene<Piece : ChessBoardState.Piece>(
    * @param pieces The new piece position
    */
   fun update(pieces: Map<Position, Piece>) {
-    savePosition(pieces)
-    moveAllPiece()
+    currentPositionChannel.trySend(pieces)
   }
 
   /**
-   * Save the newest pieces position
+   * Prepares the [ArModelNode] which contains the AR board to be displayed, by loading the
+   * appropriate model.
    *
-   * @param pieces The new pieces position to save
+   * @param node the [ArModelNode] to bee prepared.
+   * @return the [RenderableInstance] which can be used to manipulate the loaded model.
    */
-  private fun savePosition(pieces: Map<Position, Piece>) {
-    currentPosition = pieces
-  }
-
-  /** Move all pieces */
-  private fun moveAllPiece() {
-    if (isLoaded) {
-      for ((position, piece) in currentPosition) {
-        move(piece, position)
-      }
-    }
-  }
-
-  /** Move the given [piece] at a specific [position]] */
-  private fun move(piece: Piece, position: Position) {
-    val model = currentPieces[piece] ?: return
-    model.position = toArPosition(position)
-  }
+  private suspend fun prepareBoardRenderableInstance(node: ArModelNode): RenderableInstance? =
+      node.loadModel(
+          context = context,
+          glbFileLocation = ChessModels.Board,
+          autoScale = true,
+      )
 
   /**
-   * Load the board model
+   * Loads all the [Renderable] for any [Rank] and makes them available as a higher-order function.
```

Comment:
Your function is unnecessary, Sceneview have already a caching system that prevent to reload a loaded renderable

Reply:
While Sceneview indeed has a caching system for models (using a `ResourceRegistry` for renderables), accessing the registry and loading another model using the high-level `GLBLoader.loadModel` function requires an asynchronous call, since `ResourceRegistry.get()` returns a `CompletableFuture`.

When creating the `ModelNode` when the piece positions are retrieved and set, we're interested in accessing the `Renderable` without suspending execution, which can't be achieved if we call `GLBLoader.loadModel` directly.